src/main/java/org/jabref/gui/metadata/BibtexStringEditorDialogViewModel.java

```diff
@@ -59,7 +59,9 @@ public void addNewString() {
     }
 
     public void removeString() {
-        allStrings.remove(selectedItemProperty.getValue());
+        BibtexStringViewModel toBeRemoved = selectedItemProperty.getValue();
+        allStrings.remove(toBeRemoved);
+        bibDatabase.removeStringByName(fromBibtexStringViewModel(toBeRemoved));
```

Comment:
If the user now removes a string but presses `cancel` the string will still be removed, right?